src/asmcnc/comms/serial_connection.py

```diff
@@ -624,7 +624,7 @@ def cancel_stream(self):
             # The line buffer has a capacity of 35 lines
             # So the currently executing command is the one 35 lines before the last one received by the buffer
             if not self.jd.job_recovery_skip_recovery:
-                self.jd.write_to_recovery_file_after_cancel(self.g_count - 35)
+                self.jd.write_to_recovery_file_after_cancel(self.g_count - 35, self.sm.get_screen('go').time_taken_seconds)
 
```

Comment:
I would like, in future, to try and decouple the screens from the backend (I'm aware that at the moment it's a right mess and screens are pulled into the serial comms all over the place)

I think it might be better if the update_machine_runtime() function is called first, you can modify it to return time_taken_seconds at the end of the function, and then this value can be passed to the write_to_recovery function instead.